spritecontrol.cpp

```diff
@@ -12,22 +12,22 @@ SpriteGenerator::SpriteGenerator() {
 	using namespace std::chrono;
 	std::srand(duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count());
 
-	PaletteGroup palette_group = (PaletteGroup) (cfg.at(YonkPalette));
+	PaletteGroup palette_group = (PaletteGroup) (cfg[Cfg::Palette]);
 
-	std::vector<const Palette *> bag_of_palettes;
+	std::vector<const PaletteData *> bag_of_palettes;
 	if (palette_group == PaletteGroup::RandomlyGenerated) {
-		for (int i = (int) config_ranges.at(MaxColors).first; i < (int) config_ranges.at(MaxColors).second; i++) {
+		for (int i = Cfg::MaxColors.range.first; i < Cfg::MaxColors.range.second; i++) {
 			bag_of_palettes.push_back(RandomPalettes::random(i));
 		}
 	} else {
-		auto palette_names = PALETTES_BY_GROUP.at(palette_group);
-		for (auto name : palette_names) {
-			bag_of_palettes.push_back(&PALETTES.at(name));
+		auto group = PaletteGroups::get(palette_group);
+		for (const auto *palette : group.members) {
```

Comment:
This does not compile (C3535 and C2440).

Reply:
Should be fixed now.